### PR TITLE
[Migration] Redirect flow

### PR DESCRIPTION
**Pull request by waltfy**
@harrison review?

/cc @KushalP

### CODE REVIEW

File: ng-gc-security-service/security-service.js [hunk 1]
```diff
@@ -40,6 +40,16 @@ angular.module('ngSecurityService', [
...
 
+      // START: ONE-PRODUCT MIGRATION HACK
+      getLocation: function getLocation() {
+        return window.location;
+      },
+
+      setLocation: function setLocation(location) {
+        return window.location = location;
+      },
+      // END: ONE-PRODUCT MIGRATION HACK
```
Comment (KushalP) [inline ng-gc-security-service/security-service.js]: toast

**Comment by KushalP** [comment]
Given that I know a bit about how this code was written, I'm okay with us merging this in. I'll leave @harrison to finish the review.

File: ng-gc-security-service/security-service.js [hunk 2]
```diff
@@ -40,6 +40,16 @@ angular.module('ngSecurityService', [
         openSignInDialog();
       },
 
+      // START: ONE-PRODUCT MIGRATION HACK
```
Comment (waltfy) [inline ng-gc-security-service/security-service.js]: ![[as]](https://media.giphy.com/media/3Mm6HZud4m2oU/giphy.gif)

File: ng-gc-security-service/security-service.js [hunk 3]
```diff
@@ -47,9 +57,45 @@ angular.module('ngSecurityService', [
...
+              var targetHost = hostList[service.getLocation().host];
+
+              if (targetHost) {
+                service.signOut(function () {
+                  service.setLocation(
+                      template({
+                        host: targetHost,
+                        token: response.data.migration_token
+                      })
+                    );
```
Comment (harrison) [inline ng-gc-security-service/security-service.js]: indentation

File: ng-gc-security-service/security-service.spec.js [hunk 4]
```diff
@@ -62,10 +64,84 @@ describe('SecurityService', function() {
...
+      $rootScope.$digest();
+      expect(service.getLocation).not.toHaveBeenCalled();
+      expect(service.setLocation).not.toHaveBeenCalled();
+    });
+
+    describe('when redirecting to PRO for one-product', function() {
+      beforeEach(function() {
+        $window.Bugsnag = {
+          notifyException: jasmine.createSpy('notifyException'),
+        };
```
Comment (harrison) [inline ng-gc-security-service/security-service.spec.js]: `spyOn($window.Bugsnag, 'notifyException')`?

**Comment by harrison** [comment]
:shipit: :dancer:

File: ng-gc-security-service/security-service.spec.js (hunk 4 above)
Comment (waltfy) [inline ng-gc-security-service/security-service.spec.js]: `Bugsnag` is undefined at this point.

**Pull request merged**